server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java

```diff
@@ -219,10 +224,18 @@ public DiscoveryRequestStreamObserver(String defaultTypeUrl,
       ackedResources = new ConcurrentHashMap<>(Resources.TYPE_URLS.size());
       streamNonce = new AtomicLong();
       this.executor = executor;
+      validator = new ReflectiveValidatorIndex().validatorFor(DiscoveryRequest.class);
     }
 
     @Override
     public void onNext(DiscoveryRequest request) {
+      try {
+        validator.assertValid(request);
+      } catch (ValidationException e) {
+        closeWithError(ValidationExceptions.asStatus(e).asException());
```

Comment:
Would be great to have a negative test case that asserts a sample validation error.

Reply:
One thing to note AFAIK, Envoy's [DiscoveryRequest](https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/discovery.proto#L23) does not have any protobuf validations. So adding this for validating requests may not be of much help?

Reply:
Good point. Should `java-control-plane` validate responses? If so, what should it do? Log the error and continue?